pybaseball/datahelpers/caching/__init__.py

```diff
@@ -0,0 +1,117 @@
+
+            try:
+                f_hash = get_func_hash(func, args, kwargs)
+                filename = os.path.join(self.cache_config.cache_directory, f"{f_hash}.{self.extension}")
+
+                if os.path.exists(filename):
+                    modified = datetime.fromtimestamp(os.path.getmtime(filename))
+                    if (modified + self.cache_config.expiration) > datetime.now():
+                        # Hasn't expired yet
+                        if self.cache_config.profiling:
+                            start = time.perf_counter()
+                            data = self.load(filename)
+                            read_time = time.perf_counter() - start
+                        else:
+                            data = self.load(filename)
+                        if self.reset_index:
+                            data = data.reset_index(drop=True)
+                        if read_time:
+                            return data, read_time, write_time, os.path.getsize(filename)
+                        else:
+                            return data
+            except:  # pylint: disable=broad-except
+                # If this fails, log it, and then go ahead and make the function call
+                # No need to crash the real work in this phase
+                for ex in traceback.format_exc().split('\n'):
+                    logging.debug(ex)
+            result = func(*args, **kwargs)
+
+            if filename and result is not None and isinstance(result, pd.DataFrame) and not result.empty:
+                if self.reset_index:
+                    result = result.reset_index(drop=True)
+                if self.cache_config.profiling:
+                    start = time.perf_counter()
+                    self.save(result, str(filename))
+                    write_time = time.perf_counter() - start
+                else:
+                    self.save(result, filename)
+
+            if write_time and filename:
+                return result, read_time, write_time, os.path.getsize(filename)
+            else:
+                return result
+
+        if self.reset_cache:
+            self.bust_cache(func)
+
+        return _cached
+
+    def bust_cache(self, func: Callable) -> None:
+        cache_files = [
+            x for x in os.walk(self.cache_config.cache_directory).__next__()[2] if x.startswith(f"{func.__name__}(")
+        ]
+        for cache_file in cache_files:
+            _remove(os.path.join(self.cache_config.cache_directory, cache_file))
+        self.reset_cache = False
+
+    def load(self, filename: str) -> pd.DataFrame:
+        return dataframe_utils.load(filename, self.cache_config.cache_type)
+
+    def save(self, data: pd.DataFrame, filename: str) -> None:
+        dataframe_utils.save(data, filename, self.cache_config.cache_type)
+
+def bust_cache() -> None:
+    try:
+        _rmtree(cache_config.cache_directory)
+    except:
+        pass
+    _mkdir(cache_config.cache_directory)
```

Comment:
why not just
```python3
for f in os.listdir(...):
    os.remove(f)
```

Reply:
Because os.remove doe not delete subdirectories (such as lahman, or any future needs).

Reply:
Lahman should be kept separately, not in the `cache/` subdir

Reply:
So we force the user to take that, with no option to change?

Reply:
Lahman is stable. It's about a couple updates over the winter. It's not volatile the same way.

Reply:
And yeah, for this version, we force it (we already force the name in `download_lahman`), and we can throw in something else later. For simplicity's sake.

Reply:
> And yeah, for this version, we force it..and we can throw in something else later.

Do we have a compelling reason to do that? The code is written, is beneficial to the user, and works. Why throw the code away and then have to do it again later?